jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -38,22 +38,32 @@
 import org.apache.http.impl.client.DefaultHttpClient;
 
 /**
- * Thread-safe HTTP client with the automatic insecure connection failover feature. Intended to be
- * created once and shared to be called at multiple places. Callers should close the returned {@link
- * Response}.
+ * Thread-safe HTTP client that can automatically failover from secure HTTPS to insecure HTTPS or
+ * HTTP. Intended to be created once and shared to be called at multiple places. Callers should
+ * close the returned {@link Response}.
  *
  * <p>The failover (if enabled) in the following way:
  *
  * <ul>
- *   <li>When port was given:
+ *   <li>When a port is provided (for example {@code my-registry:5000/my-repo}):
  *       <ol>
  *         <li>Attempts secure HTTPS.
- *         <li>If it fails due to {@link SSLException}, attempts insecure HTTPS.
- *         <li>If it fails again due to {@link SSLException}, attempts plain-HTTP.
+ *         <li>If (1) fails due to {@link SSLException}, attempts insecure HTTPS (disables
+ *             certificate validation).
```

Comment:
Maybe _…re-attempts secure HTTPS on the specified port but disabling certificate validation_?